lte/gateway/c/session_manager/SessionStateEnforcer.h

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2020 The Magma Authors.
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*****************************************************************************
+  Source      	SessionStateEnforcer.h
+  Version     	0.1
+  Date       	2020/08/08
+  Product     	SessionD
+  Subsystem   	5G managing & maintaining state & store of session of SessionD
+                Fanout message to Access and UPF through respective client obj
+  Author/Editor Sanjay Kumar Ojha
+  Description 	Objects run in main thread context invoked by folly event
+*****************************************************************************/
+
+#pragma once
+
+#include <unordered_map>
+#include <map>
+#include <unordered_set>
+#include <vector>
+
+#include <folly/io/async/EventBaseManager.h>
+#include <lte/protos/mconfig/mconfigs.pb.h>
+#include <lte/protos/policydb.pb.h>
+#include "RuleStore.h"
+#include "PipelinedClient.h"
+#include "SessionState.h"
+#include "SessionStore.h"
+#include "AmfServiceClient.h"
+
+
+namespace magma
+{
+class SmSessionContextSendClient {};
+
+//Object flow calss for 5G and composed from 4G LocalEnforcer.
+class SessionStateEnforcer
+{
+   public:
+
+      SessionStateEnforcer(
+      std::shared_ptr<StaticRuleStore> rule_store,
+      SessionStore& session_store,
+      /*M5G specific parameter new objects to communicate UPF and response to AMF*/
+      std::shared_ptr<PipelinedClient> pipelined_client,
+      std::shared_ptr<AmfServiceClient> amf_srv_client,
+      magma::mconfig::SessionD mconfig);
+      //std::shared_ptr<SmSessionContextSendClient> SmSession_Context_Send_Client, //Revisit on sending msg back

```

Comment:
Can we remove all commented out lines unless we have clear plans to uncomment them in the future? If for some reason, we want to keep it commented out, let's add a "TODO" to make it clear that there is an action to be taken. 
For example, the `spgw_client` does not seem useful for this class.

Reply:
Done